metrics.md

```diff
@@ -0,0 +1,85 @@
+
+
+### Events
+
+Universal Search will record and report each of the following events:
+
+- User selects a recommendation.
+	- Time from search start to search completion.
+	- Length of query when selected.
+	- List of enhancers applied (e.g. `Object.keys(recommendation.enhancements)`.
+	- Value of the recommendation's `data-type` attribute. _(NB: this likely won't matter in early experiments. This is intended to allow explicit monitoring of qualitative types of results, e.g. "movie cards".)_
+- User enters a URL.
+	- Time from search start to search completion.
+- Use directly searches using their default search engine.
+	- Time from search start to search completion.
+	- Length of query.
+- User selects a suggested search term.
+	- Time from search start to search completion.
+	- Length of query.
+	- Levenstein distance between query and completed search term.
+- User selects a bookmark.
+	- Time from search start to search completion.
+	- Length of query.
+- Use selects an item from history.
+	- Time from search start to search completion.
+	- Length of query.
+
+
+### Daily ping
+
+Once a day, Universal Search will report an aggregation of activity from the previous day. That data will include:
+
+- The number of searches in the previous day.
+- The number of those searches that ended in each of the following ways:
+	- By visiting a recommendation.
+	- By entering and visiting a URL directly into the URL bar.
+	- By entering and visiting a search query into the URL bar.
+	- By using a suggested search term.
+	- By visiting a bookmark.
+	- By visting an item from history.
+	- By terminating the search without doing any of the above.
+- The number of searches for which:
+	- The client received a recommendation.
+	- A recommendation was not available.
+	- The client aborted an attempt to retrieve a recommendation
+	- There was an error while attempting to retrieve a recommendation.
+
+This data will be used in two ways:
+
+1. To gain a deeper understanding as to how users engage with the URL bar and search popup.
+2. To segment users of other metrics.
+
+
+## Server metrics
+
+To gain a better understanding of how users make use of the URL bar, the recommendation server will maintain logs of queries passed to it. For each request, we will record:
+
+- The query string itself.
+- The HTTP status code of the response.
+- The enhancers applied to that query (i.e. `recommendation.enhancements.keys()`)
+
+We do not wish to record individual users browsing habits; we're interested in aggregations of user behavior. Thus, we will not record any queries that:
+
+- Are longer than 20 characters.
+- Appear to begin with a protocol (`^[^\s]+\:\S`).
+- Appear to be a hostname (`^[^\s]+\.\S`)
+
+Additionally, on a TBD schedule we will discard outliers:
```

Comment:
I talked to Rob.  Sounds like it's fairly trivial to purge data from Redshift, and slightly more difficult to purge from S3, but we could put a retention policy on S3 since it's just a stop on the way to Redshift anyway.

tl;dr this should be possible.